src/cascadia/TerminalApp/App.cpp

```diff
@@ -176,6 +237,79 @@ namespace winrt::TerminalApp::implementation
         _ShowDialog(winrt::box_value(title), winrt::box_value(message), buttonText);
     }
 
+    // Method Description:
+    // - Displays a dialog for errors found while loading or validating the
+    //   settings. Uses the resources under the provided  title and content keys
+    //   as the title and first content of the dialog, then also displays a
+    //   message for whatever exception was found while validating the settings.
+    // - Only one dialog can be visible at a time. If another dialog is visible
+    //   when this is called, nothing happens. See _ShowDialog for details
+    // Arguments:
+    // - titleKey: The key to use to lookup the title text from our resources.
+    // - contentKey: The key to use to lookup the content text from our resources.
+    void App::_ShowLoadErrorsDialog(const winrt::hstring& titleKey,
+                                    const winrt::hstring& contentKey)
+    {
+        auto title = _resourceLoader.GetLocalizedString(titleKey);
+        auto buttonText = _resourceLoader.GetLocalizedString(L"Ok");
+
+        Controls::TextBlock warningsTextBlock;
+        // Make sure you can copy-paste
+        warningsTextBlock.IsTextSelectionEnabled(true);
+        // Make sure the lines of text wrap
+        warningsTextBlock.TextWrapping(TextWrapping::Wrap);
+
+        winrt::Windows::UI::Xaml::Documents::Run errorRun;
+        const auto errorLabel = _resourceLoader.GetLocalizedString(contentKey);
+        errorRun.Text(errorLabel);
+        warningsTextBlock.Inlines().Append(errorRun);
+
+        if (FAILED(_settingsLoadedResult))
+        {
+            if (WEB_E_INVALID_JSON_STRING == _settingsLoadedResult)
+            {
+                winrt::Windows::UI::Xaml::Documents::Run exceptionRun;
+                exceptionRun.Text(_settingsLoadExceptionText);
+                warningsTextBlock.Inlines().Append(exceptionRun);
+            }
+            else if (E_INVALIDARG == _settingsLoadedResult)

```

Comment:
these two conditionals have the same body?

Reply:
wouldn't it be better to gate on the existence of `_settingsLoadExceptionText`?

Reply:
Oh yea, those were originally more different, but I guess they're just the same now. Lets do that